tools/src/main/java/org/apache/kafka/tools/JmxTool.java

```diff
@@ -0,0 +1,434 @@
+                            if (Arrays.asList(attributesInclude.get()).contains(attribute.getName())) {
+                                expectedAttributes.add(objectName);
+                            }
+                        });
+                        if (expectedAttributes.size() > 0) {
+                            result.put(objectName, expectedAttributes.size());
+                        }
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+            } else {
+                queries.forEach(objectName -> result.put(objectName, attributesInclude.get().length));
+            }
+        }
+
+        if (result.isEmpty()) {
+            throw new TerseException(String.format("No matched attributes for the queried objects %s.", queries));
+        }
+        return result;
+    }
+
+    private static Map<String, Object> queryAttributes(MBeanServerConnection conn,
+                                                       Set<ObjectName> objectNames,
+                                                       Optional<String[]> attributesInclude) throws Exception {
+        Map<String, Object> result = new HashMap<>();
+        for (ObjectName objectName : objectNames) {
+            MBeanInfo beanInfo = conn.getMBeanInfo(objectName);
+            AttributeList attributes = conn.getAttributes(objectName,
+                    Arrays.stream(beanInfo.getAttributes()).map(a -> a.getName()).toArray(String[]::new));
+            for (Attribute attribute : attributes.asList()) {
+                if (attributesInclude.isPresent()) {
+                    if (Arrays.asList(attributesInclude.get()).contains(attribute.getName())) {
+                        result.put(String.format("%s:%s", objectName.toString(), attribute.getName()),
+                                attribute.getValue());
+                    }
+                } else {
+                    result.put(String.format("%s:%s", objectName.toString(), attribute.getName()),
+                            attribute.getValue());
+                }
+            }
+        }
+        return result;
+    }
+
+    private static void maybePrintCsvHeader(String reportFormat, List<String> keys, Map<ObjectName, Integer> numExpectedAttributes) {
+        if (reportFormat.equals("original") && keys.size() == sumValues(numExpectedAttributes) + 1) {
+            System.out.println(mkString(keys.stream().map(key -> String.format("\"%s\"", key)), ","));
+        }
+    }
+
+    private static void maybePrintDataRows(String reportFormat, Map<ObjectName, Integer> numExpectedAttributes, List<String> keys, Map<String, Object> attributes) {
+        if (attributes.size() == sumValues(numExpectedAttributes) + 1) {
+            switch (reportFormat) {
+                case "properties":
+                    keys.forEach(key -> System.out.println(String.format("%s=%s", key, attributes.get(key))));
```

Comment:
We should be able to use `System.out.printf` instead of `System.out.println(String.format`